skxray/core.py

```diff
@@ -1180,3 +1180,47 @@ def multi_tau_lags(multitau_levels, multitau_channels):
     lag_steps = np.append(lag_steps, np.array(lag))
 
     return tot_channels, lag_steps
+
+
+def circular_average(image, calibrated_center, thershold=0, nx=100,
+                     pixel_size=None):
+    """
+    Circular average(radial integration) of the intensity distribution of
+    the image data.
+
+    Parameters
+    ----------
+    image : array
+        input image
+
+    calibrated_center : tuple
+        The center in pixels-units (row, col)
+
+    thershold : float, optional
+        threshold value to mask
+
+    nx : int, optional
+        number of bins
+
+    pixel_size : tuple, optional
+        The size of a pixel in real units. (height, width). (mm)
+
+    Returns
+    -------
+    bin_centers : array
+        bin centers from bin edges
+
+    ring_averages : array
+        circular integration of SAXS intensity
+    """
+    radial_val = radial_grid(calibrated_center, image.shape,
+                                  pixel_size)
+
+    bin_edges, sums, counts = bin_1D(np.ravel(radial_val),
+                                          np.ravel(image), nx)
+    th_mask = counts > thershold
+    ring_average = sums[th_mask] / counts[th_mask]
+
+    bin_centers = bin_edges_to_centers(bin_edges)
+
+    return bin_centers, ring_average
```

Comment:
In general these are not the same length. The length of `bin_centers` will always `nx`, the length of `ring_average` will be `sum(counts > threshold)`.